src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/Snapshots/RestoreComparer.TargetFrameworkInfoEqualityComparer.cs

```diff
@@ -24,6 +24,18 @@ public override bool Equals(IVsTargetFrameworkInfo x, IVsTargetFrameworkInfo y)
                 if (!xProperties.SequenceEqual(yProperties, ProjectProperties))
                     return false;
 
+                IEnumerable<IVsReferenceItem> xFrameworkReferences = x.FrameworkReferences.Cast<IVsReferenceItem>();
+                IEnumerable<IVsReferenceItem> yFrameworkReferences = y.FrameworkReferences.Cast<IVsReferenceItem>();
+
+                if (!xFrameworkReferences.SequenceEqual(yFrameworkReferences, ReferenceItems))
+                    return false;
+
+                IEnumerable<IVsReferenceItem> xPackageDownloads = x.PackageDownloads.Cast<IVsReferenceItem>();
+                IEnumerable<IVsReferenceItem> yPackageDownloads = y.PackageDownloads.Cast<IVsReferenceItem>();
+
+                if (!xPackageDownloads.SequenceEqual(yPackageDownloads, ReferenceItems))
+                    return false;
```

Comment:
Does ordering of `FrameworkReferences` and `PackageDownloads` matter for the purposes of comparison?

Reply:
It doesn't, but I'd already used SequenceEqual here for the others as well. I do a follow up doing an unordered comparison for all of them.